packages/opentelemetry-instrumentation/src/instrumentation.ts

```diff
@@ -23,8 +23,7 @@ import * as types from './types';
 /**
  * Base abstract internal class for instrumenting node and web plugins
  */
-export abstract class InstrumentationAbstract<T = any>
-  implements types.Instrumentation {
+export abstract class InstrumentationAbstract implements types.Instrumentation {
```

Comment:
My Preference would be to keep the type so that any user can retain the type safety.

It seems to me that what we should do is (where necessary) to have the instrumentation export a "public" interface, any if the specific instrumentation doesn't want to do that, it can use "any" or "unknown".

Reply:
The type which was used here had no impact on the concrete instrumentation. The concrete type to use was (and still is) specified in the concrete implementation to instantiate the `InstrumentationNodeModuleDefinition`s.

We could keep the `<T  = any>` here but I see no value in it. Just take a look at e.g. xhr instrumenation. It has to specifies a type which is not used anywhere afterwards.

Reply:
But, for any instrumentation where they want to support their own extensions (properties or functions) this would provide additional type safety and they can choose to still hide the actual internal workings (from type safety / code completion / documentation / exported types) but provide some sort of public API etc.

Reply:
I agree, that specifying ```any``` doesn't have any advantages for what I said above. As you are basically saying that there is no type safety and do what you want. (and yes I know that is pure JS you can anyway)

Reply:
Users can (and should) still use concrete types when creating a `InstrumentationNodeModuleDefinition`.
They can even make their class extending `InstrumentationAbstract` a generic and supply a type if this helps anywhere.

But within the @opentelemetry/instrumentation module the type has no value for type safety to my understanding.

Reply:
Originally the types were added to keep consistency to what we have previously - plugins had defined type too. Some of the instrumentations already uses mixed types and quite often the last thing you can do is to use any. If that would also help to resolve issue with having to include types for the package in dependencies I would be fully into removing this anyway then.

Reply:
It seems to me this conversation is resolved no?